Manage Students: create team and display unassigned students

hirokiterashima opened the pull request:
## Changes
- Show unassigned students, and move students to/from it
- Added ability for teacher to create a new team in the new Manage Students page

## Test
- First, review and pull the PR [New Endpoint for creating new team and showing unassigned students](https://github.com/WISE-Community/WISE-API/pull/28)

## Test unassigned students
- Add a student to the run, but do not launch the unit
- As teacher, in the Manage Students view, the student should appear in the "Unassigned Students" section
- Move student to/from the "Unassigned Students" section should work

## Test create new team
- As teacher, go to the new Manage Students page. Create Team button should be there
- Click on the button. This should open a dialog for creating a new team. All of the students in the period should be listed here, including unassigned students
- Use the dialog to choose initial members for the new team. If the student is already in a team, a warning should appear.
 
Closes #308

hirokiterashima committed:
Added CreateTeam in ManageStudents.
Also show unassigned students in ManageStudents, and allow moving between workgroups and unassigned.

#308

geoffreykwan reviewed (changes requested):
The functionality works well. I found a bug where if there is an unassigned student, the "Grade By Team" view will throw an error and display an empty workgroup row.

```
TypeError: Cannot read property 'length' of undefined
    at WorkgroupSelectAutocompleteComponent.getStudentsFromWorkgroups (workgroup-select-autocomplete.component.ts:74)
    at WorkgroupSelectAutocompleteComponent.setWorkgroups (workgroup-select-autocomplete.component.ts:62)
    at WorkgroupSelectAutocompleteComponent.ngOnInit (workgroup-select.component.ts:26)
    at WorkgroupSelectAutocompleteComponent.ngOnInit (workgroup-select-autocomplete.component.ts:29)
```

The null pointer error is caused by userIds being null in the object that contains the unassigned students.

https://github.com/WISE-Community/WISE-Client/blob/69315c5c24c669ec0e0447d917e73c55f8ce7a37/src/app/classroom-monitor/workgroup-select/workgroup-select-autocomplete/workgroup-select-autocomplete.component.ts#L69-L86

![Screenshot 2021-08-30 122711](https://user-images.githubusercontent.com/1036509/131372251-65e48365-64e1-4eb2-ac15-42b1e9fe94b9.jpg)


- If you create a new team using a student that is already in a team, should the popup display the student names and show which ones are already in a team and which are unassigned?

- Should we allow moving an unassigned student to a different period?

hirokiterashima committed:
Hide unassigned student from grading views.
Refactored isWorkgroupShown function.

hirokiterashima committed:
Added warn class to unassigned students.

breity committed:
Updated some styles and layout for Manage Students:
- Made unassigned students container full width, added columns.
- Changed color of unassigned students to 'warn'.
- Added outline, 'primary' color, and changed location of new team button.
#308

breity committed:
Don't allow moving students from teams to unassigned students. #308

breity committed:
Fixed bug that prevented moving student to new team.
Only occurerd when All Periods was selected in Teacher Tools.
#308

breity committed:
Fixed bug that prevented moving student to new team.
Only occurerd when All Periods was selected in Teacher Tools.
Fixed tests.
#308

hirokiterashima committed:
Hide unassigned students legend if there are no unassigned students. #308

hirokiterashima committed:
Hide warning text in move student dialog when moving unassigned student to a team. #308

hirokiterashima committed:
Hide new team button if there are no students in the period. #308

breity reviewed (approved):
Looks good. I styled and made UI changes we discussed.

geoffreykwan reviewed (approved):
Looks good.
